internal/gatewayapi/route.go

```diff
@@ -1226,16 +1226,68 @@ func (t *Translator) processDestination(backendRefContext BackendRefContext,
 			"Mixed endpointslice address type for the same backendRef is not supported")
 	}
 
+	destinationFilters := t.processDestinationFilters(routeType, backendRefContext, parentRef, route, resources)
 	ds = &ir.DestinationSetting{
 		Weight:      &weight,
 		Protocol:    protocol,
 		Endpoints:   endpoints,
 		AddressType: addrType,
 		TLS:         backendTLS,
+		Filters:     destinationFilters,
 	}
+
 	return ds
 }
 
+func getBackendFilters(routeType gwapiv1.Kind, backendRefContext BackendRefContext) (backendFilters any) {
+	switch routeType {
+	case KindHTTPRoute:
+		backendFilters = GetFilters(backendRefContext).([]gwapiv1.HTTPRouteFilter)
+	case KindGRPCRoute:
+		backendFilters = GetFilters(backendRefContext).([]gwapiv1.GRPCRouteFilter)
+	}
+	return
+}
+
+func (t *Translator) processDestinationFilters(routeType gwapiv1.Kind, backendRefContext BackendRefContext, parentRef *RouteParentContext, route RouteContext, resources *Resources) *ir.DestinationFilters {
+	backendFilters := getBackendFilters(routeType, backendRefContext)
+	if backendFilters == nil {
+		return nil
+	}
+	var httpFilters []gwapiv1.HTTPRouteFilter
+	var grpcFilters []gwapiv1.GRPCRouteFilter
+	var httpFiltersContext *HTTPFiltersContext
+	var destFilters ir.DestinationFilters
+
+	switch filters := backendFilters.(type) {
+	case []gwapiv1.HTTPRouteFilter:
+		httpFilters = append(httpFilters, filters...)
```

Comment:
why this is additional variable needed ?

Reply:
not at all, thanks